apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/ErrorLoggingListener.java

```diff
@@ -35,7 +35,7 @@ public void onError(String typeName, ClassLoader classLoader, JavaModule module,
                 typeName,
                 ((MinimumClassFileVersionValidator.UnsupportedClassFileVersionException)throwable).getMinVersion());
         } else {
-            if (throwable.getMessage().contains("Cannot resolve type description")) {
+            if (throwable != null && throwable.getMessage() != null && throwable.getMessage().contains("Cannot resolve type description")) {
```

Comment:
It's good to guard but if the throwable is `null`, it may be a bug and even a `null` message is weird.
@raphw are you aware of specific scenario where this happens with `net.bytebuddy.agent.builder.AgentBuilder.Listener.Adapter#onError`?

Reply:
See reply on thread. Beat to replace with an instanceof check.

Reply:
```suggestion
            if (throwable instanceof net.bytebuddy.pool.TypePool$Resolution$NoSuchTypeException) {
```

This used to be a plain `IllegalStateException` in previous versions of Byte Buddy which is why we had to check the message. Now, there's a dedicated exception type that extends `IllegalStateException`.